src/netbase.cpp

```diff
@@ -161,32 +161,31 @@ static std::vector<CNetAddr> LookupIntern(const std::string& name, unsigned int
     return addresses;
 }
 
-bool LookupHost(const std::string& name, std::vector<CNetAddr>& addresses, unsigned int nMaxSolutions, bool fAllowLookup, DNSLookupFn dns_lookup_function)
+std::vector<CNetAddr> LookupHost(const std::string& name, unsigned int nMaxSolutions, bool fAllowLookup, DNSLookupFn dns_lookup_function)
 {
+    std::vector<CNetAddr> addresses;
     if (!ContainsNoNUL(name)) {
-        return false;
+        return addresses;
     }
     std::string strHost = name;
     if (strHost.empty())
-        return false;
+        return addresses;
```

Comment:
nit: the `addresses` vector is not needed anymore in the first `LookupHost` function and can be removed:
```suggestion
    if (!ContainsNoNUL(name)) return {};
    std::string strHost = name;
    if (strHost.empty()) return {};
```

Reply:
Nice, addressed!